src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -2317,6 +2322,9 @@ int xcldev::device::testM2m()
     }
 
     for(int32_t i = 0; i < map->m_count; i++) {
+        if (isHostMem((const char *)map->m_mem_data[i].m_tag))
```

Comment:
why not use std::compare to compare the string instead of converting it to const char* first?

Reply:
This is a C interface as "unsigned char". If we use CPP compare, we have to add more code like std::string(m_tag) then compare. 

I am ok to make the changes.

Reply:
Yes, let's use cpp :)

Reply:
Do not convert unnecessarily to std::string.  I believe isHostMem is wasteful (const char* -> string -> const char*) strncmp would be better in this case